<?xml version="1.0" ?>
<sdf version="1.8">
  <world name="cow_drone_world">
    <physics name="default_physics" type="ode">
      <gravity>0 0 -9.8</gravity>
      <max_step_size>0.01</max_step_size>
      <real_time_update_rate>1000</real_time_update_rate>
    </physics>
    <plugin filename="libignition-gazebo-physics-system.so" name="ignition::gazebo::systems::Physics"></plugin>
    <plugin filename="libignition-gazebo-user-commands-system.so" name="ignition::gazebo::systems::UserCommands"></plugin>
    <plugin filename="libignition-gazebo-scene-broadcaster-system.so" name="ignition::gazebo::systems::SceneBroadcaster"></plugin>
    <light type="directional" name="sun">
      <cast_shadows>true</cast_shadows>
      <pose>0 0 10 0 0 0</pose>
      <diffuse>0.8 0.8 0.8 1</diffuse>
      <specular>0.2 0.2 0.2 1</specular>
      <attenuation>
        <range>1000</range>
        <constant>0.9</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <direction>-0.5 0.1 -0.9</direction>
    </light>

    <model name="ground_plane">
      <static>true</static>
      <link name="link">
        <collision name="collision">
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>100 100</size>
            </plane>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>100 100</size>
            </plane>
          </geometry>
          <material>
            <ambient>0.8 0.8 0.8 1</ambient>
            <diffuse>0.8 0.8 0.8 1</diffuse>
            <specular>0.8 0.8 0.8 1</specular>
          </material>
        </visual>
      </link>
    </model>

    <!-- Cow (Sphere) -->
    <model name="cow1">
      <pose>0 2 0.5 0 0 0</pose>
      <link name="body">
        <inertial>
          <mass>10.0</mass>
          <inertia>
            <ixx>1.0</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>1.0</iyy>
            <iyz>0</iyz>
            <izz>1.0</izz>
          </inertia>
        </inertial>
        <collision name="collision">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
          <material>
            <ambient>0.9 0.7 0.5 1</ambient>
            <diffuse>0.9 0.7 0.5 1</diffuse>
            <specular>0.9 0.7 0.5 1</specular>
          </material>
        </visual>
      </link>

        <!-- Add a velocity controller plugin -->
      <plugin filename="libignition-gazebo-velocity-control-system.so" 
            name="ignition::gazebo::systems::VelocityControl">
        <topic>/cow1/cmd_vel</topic>
      </plugin>

      <plugin filename="libignition-gazebo-odometry-publisher-system.so"
            name="ignition::gazebo::systems::OdometryPublisher">
        <odom_frame>cow1/odom</odom_frame>
        <robot_base_frame>cow1</robot_base_frame>
        <odom_publish_frequency>10</odom_publish_frequency>
        <odom_topic>/cow1/odom</odom_topic>
      </plugin>
    </model>

    <!-- Cow (Sphere) -->
    <model name="cow2">
      <pose>5 -2 0.5 0 0 0</pose>
      <link name="body">
        <inertial>
          <mass>10.0</mass>
          <inertia>
            <ixx>1.0</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>1.0</iyy>
            <iyz>0</iyz>
            <izz>1.0</izz>
          </inertia>
        </inertial>
        <collision name="collision">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
          <material>
            <ambient>0.9 0.7 0.5 1</ambient>
            <diffuse>0.9 0.7 0.5 1</diffuse>
            <specular>0.9 0.7 0.5 1</specular>
          </material>
        </visual>
      </link>
        <!-- Add a velocity controller plugin -->
      <plugin filename="libignition-gazebo-velocity-control-system.so" 
            name="ignition::gazebo::systems::VelocityControl">
        <topic>/cow2/cmd_vel</topic>
      </plugin>

      <plugin filename="libignition-gazebo-odometry-publisher-system.so"
            name="ignition::gazebo::systems::OdometryPublisher">
        <odom_frame>cow2/odom</odom_frame>
        <robot_base_frame>cow2</robot_base_frame>
        <odom_publish_frequency>10</odom_publish_frequency>
        <odom_topic>/cow2/odom</odom_topic>
      </plugin>
    </model>


    <!-- Cow (Sphere) -->
    <model name="cow3">
      <pose>5 -2 0.5 0 0 0</pose>
      <link name="body">
        <inertial>
          <mass>10.0</mass>
          <inertia>
            <ixx>1.0</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>1.0</iyy>
            <iyz>0</iyz>
            <izz>1.0</izz>
          </inertia>
        </inertial>
        <collision name="collision">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
          <material>
            <ambient>0.9 0.7 0.5 1</ambient>
            <diffuse>0.9 0.7 0.5 1</diffuse>
            <specular>0.9 0.7 0.5 1</specular>
          </material>
        </visual>
      </link>
        <!-- Add a velocity controller plugin -->
      <plugin filename="libignition-gazebo-velocity-control-system.so" 
            name="ignition::gazebo::systems::VelocityControl">
        <topic>/cow3/cmd_vel</topic>
      </plugin>
      <plugin filename="libignition-gazebo-odometry-publisher-system.so"
            name="ignition::gazebo::systems::OdometryPublisher">
        <odom_frame>cow3/odom</odom_frame>
        <robot_base_frame>cow3</robot_base_frame>
        <odom_publish_frequency>10</odom_publish_frequency>
        <odom_topic>/cow3/odom</odom_topic>
      </plugin>
    </model>


    <!-- Cow (Sphere) -->
    <model name="cow4">
      <pose>3 -3 0.5 0 0 0</pose>
      <link name="body">
        <inertial>
          <mass>10.0</mass>
          <inertia>
            <ixx>1.0</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>1.0</iyy>
            <iyz>0</iyz>
            <izz>1.0</izz>
          </inertia>
        </inertial>
        <collision name="collision">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
          <material>
            <ambient>0.9 0.7 0.5 1</ambient>
            <diffuse>0.9 0.7 0.5 1</diffuse>
            <specular>0.9 0.7 0.5 1</specular>
          </material>
        </visual>
      </link>
        <!-- Add a velocity controller plugin -->
      <plugin filename="libignition-gazebo-velocity-control-system.so" 
            name="ignition::gazebo::systems::VelocityControl">
        <topic>/cow4/cmd_vel</topic>
      </plugin>
      <plugin filename="libignition-gazebo-odometry-publisher-system.so"
            name="ignition::gazebo::systems::OdometryPublisher">
        <odom_frame>cow4/odom</odom_frame>
        <robot_base_frame>cow4</robot_base_frame>
        <odom_publish_frequency>10</odom_publish_frequency>
        <odom_topic>/cow4/odom</odom_topic>
      </plugin>
    </model>


    <!-- Cow (Sphere) -->
    <model name="cow5">
      <pose>6 -1 0.5 0 0 0</pose>
      <link name="body">
        <inertial>
          <mass>10.0</mass>
          <inertia>
            <ixx>1.0</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>1.0</iyy>
            <iyz>0</iyz>
            <izz>1.0</izz>
          </inertia>
        </inertial>
        <collision name="collision">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
          <material>
            <ambient>0.9 0.7 0.5 1</ambient>
            <diffuse>0.9 0.7 0.5 1</diffuse>
            <specular>0.9 0.7 0.5 1</specular>
          </material>
        </visual>
      </link>
        <!-- Add a velocity controller plugin -->
      <plugin filename="libignition-gazebo-velocity-control-system.so" 
            name="ignition::gazebo::systems::VelocityControl">
        <topic>/cow5/cmd_vel</topic>
      </plugin>

      <plugin filename="libignition-gazebo-odometry-publisher-system.so"
            name="ignition::gazebo::systems::OdometryPublisher">
        <odom_frame>cow5/odom</odom_frame>
        <robot_base_frame>cow5</robot_base_frame>
        <odom_publish_frequency>10</odom_publish_frequency>
        <odom_topic>/cow5/odom</odom_topic>
      </plugin>
    </model>



    <!-- Cow (Sphere) -->
    <model name="cow6">
      <pose>6 2 0.5 0 0 0</pose>
      <link name="body">
        <inertial>
          <mass>10.0</mass>
          <inertia>
            <ixx>1.0</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>1.0</iyy>
            <iyz>0</iyz>
            <izz>1.0</izz>
          </inertia>
        </inertial>
        <collision name="collision">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
          <material>
            <ambient>0.9 0.7 0.5 1</ambient>
            <diffuse>0.9 0.7 0.5 1</diffuse>
            <specular>0.9 0.7 0.5 1</specular>
          </material>
        </visual>
      </link>
        <!-- Add a velocity controller plugin -->
      <plugin filename="libignition-gazebo-velocity-control-system.so" 
            name="ignition::gazebo::systems::VelocityControl">
        <topic>/cow6/cmd_vel</topic>
      </plugin>
      <plugin filename="libignition-gazebo-odometry-publisher-system.so"
            name="ignition::gazebo::systems::OdometryPublisher">
        <odom_frame>cow6/odom</odom_frame>
        <robot_base_frame>cow6</robot_base_frame>
        <odom_publish_frequency>10</odom_publish_frequency>
        <odom_topic>/cow6/odom</odom_topic>
      </plugin>
    </model>



    <!-- Cow (Sphere) -->
    <model name="cow7">
      <pose>0 -2 0.5 0 0 0</pose>
      <link name="body">
        <inertial>
          <mass>10.0</mass>
          <inertia>
            <ixx>1.0</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>1.0</iyy>
            <iyz>0</iyz>
            <izz>1.0</izz>
          </inertia>
        </inertial>
        <collision name="collision">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
          <material>
            <ambient>0.9 0.7 0.5 1</ambient>
            <diffuse>0.9 0.7 0.5 1</diffuse>
            <specular>0.9 0.7 0.5 1</specular>
          </material>
        </visual>
      </link>

        <!-- Add a velocity controller plugin -->
      <plugin filename="libignition-gazebo-velocity-control-system.so" 
            name="ignition::gazebo::systems::VelocityControl">
        <topic>/cow7/cmd_vel</topic>
      </plugin>

      <plugin filename="libignition-gazebo-odometry-publisher-system.so"
            name="ignition::gazebo::systems::OdometryPublisher">
        <odom_frame>cow7/odom</odom_frame>
        <robot_base_frame>cow7</robot_base_frame>
        <odom_publish_frequency>10</odom_publish_frequency>
        <odom_topic>/cow7/odom</odom_topic>
      </plugin>
    </model>

    <!-- Cow (Sphere) -->
    <model name="cow8">
      <pose>-5 -2 0.5 0 0 0</pose>
      <link name="body">
        <inertial>
          <mass>10.0</mass>
          <inertia>
            <ixx>1.0</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>1.0</iyy>
            <iyz>0</iyz>
            <izz>1.0</izz>
          </inertia>
        </inertial>
        <collision name="collision">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
          <material>
            <ambient>0.9 0.7 0.5 1</ambient>
            <diffuse>0.9 0.7 0.5 1</diffuse>
            <specular>0.9 0.7 0.5 1</specular>
          </material>
        </visual>
      </link>
        <!-- Add a velocity controller plugin -->
      <plugin filename="libignition-gazebo-velocity-control-system.so" 
            name="ignition::gazebo::systems::VelocityControl">
        <topic>/cow8/cmd_vel</topic>
      </plugin>

      <plugin filename="libignition-gazebo-odometry-publisher-system.so"
            name="ignition::gazebo::systems::OdometryPublisher">
        <odom_frame>cow8/odom</odom_frame>
        <robot_base_frame>cow8</robot_base_frame>
        <odom_publish_frequency>10</odom_publish_frequency>
        <odom_topic>/cow8/odom</odom_topic>
      </plugin>
    </model>


    <!-- Cow (Sphere) -->
    <model name="cow9">
      <pose>-5 -2 0.5 0 0 0</pose>
      <link name="body">
        <inertial>
          <mass>10.0</mass>
          <inertia>
            <ixx>1.0</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>1.0</iyy>
            <iyz>0</iyz>
            <izz>1.0</izz>
          </inertia>
        </inertial>
        <collision name="collision">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
          <material>
            <ambient>0.9 0.7 0.5 1</ambient>
            <diffuse>0.9 0.7 0.5 1</diffuse>
            <specular>0.9 0.7 0.5 1</specular>
          </material>
        </visual>
      </link>
        <!-- Add a velocity controller plugin -->
      <plugin filename="libignition-gazebo-velocity-control-system.so" 
            name="ignition::gazebo::systems::VelocityControl">
        <topic>/cow9/cmd_vel</topic>
      </plugin>

      <plugin filename="libignition-gazebo-odometry-publisher-system.so"
            name="ignition::gazebo::systems::OdometryPublisher">
        <odom_frame>cow9/odom</odom_frame>
        <robot_base_frame>cow9</robot_base_frame>
        <odom_publish_frequency>10</odom_publish_frequency>
        <odom_topic>/cow9/odom</odom_topic>
      </plugin>
    </model>


    <!-- Cow (Sphere) -->
    <model name="cow10">
      <pose>-3 -3 0.5 0 0 0</pose>
      <link name="body">
        <inertial>
          <mass>10.0</mass>
          <inertia>
            <ixx>1.0</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>1.0</iyy>
            <iyz>0</iyz>
            <izz>1.0</izz>
          </inertia>
        </inertial>
        <collision name="collision">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
          <material>
            <ambient>0.9 0.7 0.5 1</ambient>
            <diffuse>0.9 0.7 0.5 1</diffuse>
            <specular>0.9 0.7 0.5 1</specular>
          </material>
        </visual>
      </link>
        <!-- Add a velocity controller plugin -->
      <plugin filename="libignition-gazebo-velocity-control-system.so" 
            name="ignition::gazebo::systems::VelocityControl">
        <topic>/cow10/cmd_vel</topic>
      </plugin>

      <plugin filename="libignition-gazebo-odometry-publisher-system.so"
            name="ignition::gazebo::systems::OdometryPublisher">
        <odom_frame>cow10/odom</odom_frame>
        <robot_base_frame>cow10</robot_base_frame>
        <odom_publish_frequency>10</odom_publish_frequency>
        <odom_topic>/cow10/odom</odom_topic>
      </plugin>
    </model>


    <!-- Cow (Sphere) -->
    <model name="cow11">
      <pose>-6 -1 0.5 0 0 0</pose>
      <link name="body">
        <inertial>
          <mass>10.0</mass>
          <inertia>
            <ixx>1.0</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>1.0</iyy>
            <iyz>0</iyz>
            <izz>1.0</izz>
          </inertia>
        </inertial>
        <collision name="collision">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
          <material>
            <ambient>0.9 0.7 0.5 1</ambient>
            <diffuse>0.9 0.7 0.5 1</diffuse>
            <specular>0.9 0.7 0.5 1</specular>
          </material>
        </visual>
      </link>
        <!-- Add a velocity controller plugin -->
      <plugin filename="libignition-gazebo-velocity-control-system.so" 
            name="ignition::gazebo::systems::VelocityControl">
        <topic>/cow11/cmd_vel</topic>
      </plugin>

      <plugin filename="libignition-gazebo-odometry-publisher-system.so"
            name="ignition::gazebo::systems::OdometryPublisher">
        <odom_frame>cow11/odom</odom_frame>
        <robot_base_frame>cow11</robot_base_frame>
        <odom_publish_frequency>10</odom_publish_frequency>
        <odom_topic>/cow11/odom</odom_topic>
      </plugin>
    </model>



    <!-- Cow (Sphere) -->
    <model name="cow12">
      <pose>-6 2 0.5 0 0 0</pose>
      <link name="body">
        <inertial>
          <mass>10.0</mass>
          <inertia>
            <ixx>1.0</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>1.0</iyy>
            <iyz>0</iyz>
            <izz>1.0</izz>
          </inertia>
        </inertial>
        <collision name="collision">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <cylinder>
              <radius>0.25</radius>
              <length>0.25</length>
            </cylinder>
          </geometry>
          <material>
            <ambient>0.9 0.7 0.5 1</ambient>
            <diffuse>0.9 0.7 0.5 1</diffuse>
            <specular>0.9 0.7 0.5 1</specular>
          </material>
        </visual>
      </link>
        <!-- Add a velocity controller plugin -->
      <plugin filename="libignition-gazebo-velocity-control-system.so" 
            name="ignition::gazebo::systems::VelocityControl">
        <topic>/cow12/cmd_vel</topic>
      </plugin>

      <plugin filename="libignition-gazebo-odometry-publisher-system.so"
            name="ignition::gazebo::systems::OdometryPublisher">
        <odom_frame>cow12/odom</odom_frame>
        <robot_base_frame>cow12</robot_base_frame>
        <odom_publish_frequency>10</odom_publish_frequency>
        <odom_topic>/cow12/odom</odom_topic>
      </plugin>
    </model>
    
    

    <!-- Drone (Cylinder) -->
<model name="drone">
  <pose>0 0 5.0 0 0 0</pose>
  <link name="body">
    <inertial>
      <mass>1.0</mass>
      <inertia>
        <ixx>0.1</ixx>
        <ixy>0</ixy>
        <ixz>0</ixz>
        <iyy>0.1</iyy>
        <iyz>0</iyz>
        <izz>0.1</izz>
      </inertia>
    </inertial>
    <collision name="collision">
      <geometry>
        <cylinder>
          <radius>0.2</radius>
          <length>0.1</length>
        </cylinder>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <cylinder>
          <radius>0.2</radius>
          <length>0.1</length>
        </cylinder>
      </geometry>
      <material>
        <ambient>0.2 0.2 0.2 1</ambient>
        <diffuse>0.2 0.2 0.2 1</diffuse>
        <specular>0.2 0.2 0.2 1</specular>
      </material>
    </visual>
  </link>

  </model>
  </world>
</sdf>